BiometricLoginKotlin/app/src/main/java/com/example/biometricloginsample/LoginActivity.kt

```diff
@@ -92,7 +89,7 @@ class LoginActivity : AppCompatActivity() {
         if (canAuthenticate == BiometricManager.BIOMETRIC_SUCCESS) {
             val secretKeyName = getString(R.string.secret_key_name)
             val cipher = cryptographyManager.getInitializedCipherForDecryption(
-                secretKeyName, encryptedServerTokenWrapper.initializationVector
+                secretKeyName, ciphertextWrapper!!.initializationVector
```

Comment:
Can this ever be null? If it is, do you want to crash?

Reply:
it can never be null

Reply:
In that case, why is it nullable?

Reply:
ah, now I understand the question. The property itself is nullable. Normally I don't like `!!`, but went with it here since the use-case is straightforward and at this point will not actually crash. But I'll rewrite it out.